app.go

```diff
@@ -464,7 +529,12 @@ func New(opts ...Option) *App {
 	app.provide(provide{Target: app.dotGraph, Stack: frames})
 
 	if app.err != nil {
-		app.logger.Printf("Error after options were applied: %v", app.err)
+		fxlog.Info(
+			"error encountered while applying options",
+			fxlog.Field{
+				Key:   "error",
+				Value: app.err,
+			}).Write(app.log)
```

Comment:
You may optionally define a top-level `Err` function in the fxlog package that builds an `fxlog.Field` with the key "error"

```
func Err(err error) Field
```

Or better yet, instead of specifying `fxlog.Field{Key: ..., Value: ...}` everywhere, we can also switch to an `fxlog.F` helper function:

```
func F(string, interface{}) Field
```

It doesn't matter much because it's all internal—it's just a matter of how usable we find it inside Fx.